src/hardware/mixer.cpp

```diff
@@ -260,6 +260,25 @@ void MixerChannel::Enable(const bool should_enable)
 	MIXER_UnlockAudioDevice();
 }
 
+void MixerChannel::ConfigureResampler()
+{
+	const auto in_rate = zoh_upsampler.enabled ? zoh_upsampler.target_freq
+	                                           : sample_rate;
+	const auto out_rate = mixer.freq;
```

Comment:
There's a bunch of places now where we carry the (poorly named) `mixer.freq` value into some channel code and name is more consistently as `out_rate`. 

Maybe we can replace it all in one go with a permanent reference inside the channel itself:


``` diff
diff --git a/include/mixer.h b/include/mixer.h
index 3929042e..c3f1a328 100644
--- a/include/mixer.h
+++ b/include/mixer.h
@@ -169,6 +169,7 @@ private:
        // >0. Still work in progress and thus disabled for now.
        int offset[2] = {0, 0};
        int sample_rate = 0u;
+       const int &out_rate;
        float volmul[2] = {1.0f, 1.0f};
        float scale[2] = {1.0f, 1.0f};
 
diff --git a/src/hardware/mixer.cpp b/src/hardware/mixer.cpp
index 309a70f1..af849caf 100644
--- a/src/hardware/mixer.cpp
+++ b/src/hardware/mixer.cpp
@@ -117,7 +117,10 @@ static struct mixer_t mixer = {};
 
 uint8_t MixTemp[MIXER_BUFSIZE] = {};
 
-MixerChannel::MixerChannel(MIXER_Handler _handler, const char *_name) : envelope(_name), handler(_handler)
+MixerChannel::MixerChannel(MIXER_Handler _handler, const char *_name)
+        : envelope(_name),
+          handler(_handler),
+          out_rate(mixer.freq)
 {}
```

Because it's a `const &`, it will alway represent the `mixer.freq` value and can never point to another variable.

Reply:
> There's a bunch of places now where we carry the (poorly named) `mixer.freq` value into the channel code and give it the more consistent `out_rate` name.

`out_rate` is only used in `MixerChannel::ConfigureResampler`. I think `in_rate` and `out_rate` make perfect sense in the context of this resampler method:

```cpp
	const auto in_rate = zoh_upsampler.enabled ? zoh_upsampler.target_freq
	                                           : sample_rate;
	const auto out_rate = mixer.freq;
```

`mixer.freq` is renamed to `sample_rate` in `ConfigureLowPassFilter`; that name makes perfect sense in _that_ context:

```
void MixerChannel::ConfigureLowPassFilter(const uint8_t order,
                                          const uint16_t cutoff_freq)
{
	assert(order > 0 && order <= max_filter_order);
	const auto sample_rate = mixer.freq;
	for (auto i = 0; i < 2; ++i)
		filter.lpf[i].setup(order, sample_rate, cutoff_freq);
}
```

How I see it, `in_rate` and `out_rate` in the resampler are fine and clarify the intent. If anything needs to be changed that would be `mixer.freq` -- frequency of what, exactly? I'd refactor that to `mixer.sample_rate`, then it would be consistent with `MixerChannel` that already has a `sample_rate` member var. While we're at it, I'd also refactor the slightly nonsensical `MixerChannel::SetFreq` to `Mixer::SetSampleRate`.

Reply:
> How I see it, in_rate and out_rate in the resampler are fine and clarify the intent.

Yup - same here.

> If anything needs to be changed that would be mixer.freq -- frequency of what, exactly? I'd refactor that to mixer.sample_rate, then it would be consistent with MixerChannel that already has a sample_rate member var. While we're at it, I'd also refactor the slightly nonsensical MixerChannel::SetFreq to Mixer::SetSampleRate.

Roger that!